Beybi/Screens/ViewMorePageView/ViewMoreViewModel.swift

```diff
@@ -0,0 +1,197 @@
+                    print(food)
+                    self.foodArray?.append(food)
+                }
+                self.delegate?.reloadData()
+            }
+        }
+    }
+    
+    func readMainDishes() {
+        firestore.collection("main dishes").getDocuments { (querySnapshot, error) in
+            if let error = error {
+                print("Hata: \(error.localizedDescription)")
+            } else {
+                for document in querySnapshot!.documents {
+                    let data = document.data()
+                    let name = data["name"] as? String ?? "İsim yok"
+                    let cookingTime = data["cooking time"] as? String ?? "35-40 min"
+                    let recipe = data["recipe"] as? String ?? "Tarif yok"
+                    let imageUrl = data["imageUrl"] as? String ?? "Foto yok"
+                    let type = data["type"] as? String ?? "Type yok"
+                    let introText = data["introText"] as? String ?? "intro text yok"
+                    let ingredients = data["ingredients"] as? [String] ?? ["Malzeme yok"]
+                    let recipeStep = data["recipeStep"] as? [String] ?? ["Tarif yok"]
+                    
+                    let food = Food(name: name, cookingTime: cookingTime, recipe: recipe, imageUrl: imageUrl, type: type, introText: introText, ingredients: ingredients, recipeStep: recipeStep)
+                    self.foodArray?.append(food)
+                }
+                self.delegate?.reloadData()
+            }
+        }
+    }
+    
+    func readPurees() {
+        firestore.collection("purees").getDocuments { (querySnapshot, error) in
+            if let error = error {
+                print("Hata: \(error.localizedDescription)")
+            } else {
+                for document in querySnapshot!.documents {
+                    let data = document.data()
+                    let name = data["name"] as? String ?? "İsim yok"
+                    let cookingTime = data["cooking time"] as? String ?? "Süre yok"
+                    let recipe = data["recipe"] as? String ?? "Tarif yok"
+                    let imageUrl = data["imageUrl"] as? String ?? "Foto yok"
+                    let type = data["type"] as? String ?? "Type yok"
+                    let introText = data["introText"] as? String ?? "intro text yok"
+                    let ingredients = data["ingredients"] as? [String] ?? ["Malzeme yok"]
+                    let recipeStep = data["recipeStep"] as? [String] ?? ["Tarif yok"]
+                    
+                    let food = Food(name: name, cookingTime: cookingTime, recipe: recipe, imageUrl: imageUrl, type: type, introText: introText, ingredients: ingredients, recipeStep: recipeStep)
+                    self.foodArray?.append(food)
+                }
+                self.delegate?.reloadData()
+            }
+        }
+    }
+    
+    func readSnacks() {
+        firestore.collection("snacks").getDocuments { (querySnapshot, error) in
```

Comment:
These lines are same in almost every file. Maybe you can merge them and create a common helper